File: csharp/fundamentals/human/Human.cs
namespace Human{


    public class Human 
    {   
        public Human(){
            Strength = 3;
            Intelligence = 3;
            Dexerity = 3;
            Health = 100;

        }

        public Human(string name){
            Name = name;
        }

        public Human(string name, int strength, int intelligence, int dexerity, int health)
        {
            Name = name;
            Strength = strength;
            Intelligence = intelligence;
            Dexerity = dexerity;
            Health = health;
        }

        public string Name {get; set;}
        public int Strength {get; set;}

        public int Intelligence {get; set;}
        public int Dexerity {get; set;}
        public int Health {get; set;}

        public void acttack(object opponent){
            if (opponent is Human) {
                int damage = 5 * Strength;
                ((Human)opponent).Health =- damage;
            }
        }

    }

}

File: ruby/first_vargrant_box/oop/human/human.rb
class Human
    attr_accessor:strength,:intelligence,:stealth,:health

    def initialize 
        @strength = 3
        @stealth = 3
        @intelligence = 3
        @health = 100
    end

    def attack victim, times=10

        if victim.class.ancestors.include?  Human
            unless victim.health < 1*times
                victim.health -=1*times
            end  
            return true      
        end
        return false
            
    end
end


h = Human.new
h2 = Human.new
puts h.attack(h2)
puts h.attack("Not a Human")
puts h2.health